src/sys/socket/mod.rs

```diff
@@ -1202,3 +1206,70 @@ pub fn shutdown(df: RawFd, how: Shutdown) -> Result<()> {
         Errno::result(shutdown(df, how)).map(drop)
     }
 }
+
+#[cfg(any(
+target_os = "linux",
+))]
+#[repr(C)]
+#[allow(missing_debug_implementations)]
+pub struct MMsgHdr<'a>(libc::mmsghdr, PhantomData<&'a ()>);
+
+#[cfg(any(
+target_os = "linux",
+))]
+impl<'a> MMsgHdr<'a> {
+    pub fn new(iov: &mut[IoVec<&'a mut [u8]>], flags: MsgFlags) -> MMsgHdr<'a> {
+        // TODO:
+        //  - support 'control'
+        //  - support 'name'
+        let mut hdr: libc::mmsghdr = unsafe { mem::uninitialized() };
+        hdr.msg_hdr.msg_control = ptr::null_mut();
+        hdr.msg_hdr.msg_controllen = 0;
+        hdr.msg_hdr.msg_flags = flags.bits();
+        hdr.msg_hdr.msg_iov = iov.as_ptr() as *mut libc::iovec;
+        hdr.msg_hdr.msg_iovlen = iov.len() as _;
+        hdr.msg_hdr.msg_name = ptr::null_mut();
+        hdr.msg_hdr.msg_namelen = 0;
+        hdr.msg_len = 0;
+        MMsgHdr(hdr, PhantomData)
+    }
+
+    pub fn msg_len(&self) -> usize {
+        self.0.msg_len as usize
+    }
+}
+
+#[cfg(any(
+target_os = "linux",
```

Comment:
sendmmsg and recvmmsg are not unique to Linux.  You should enable them on all platforms that support them.

Reply:
I've added freebsd and netbsd.  Let me know if I missed something.

Reply:
Actually netbsd and freebsd builds failed e.g. https://travis-ci.org/nix-rust/nix/jobs/484619844 , since the libc crate doesn't define these.

So removed those again.